server/broadcast.go

```diff
@@ -396,23 +402,23 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string)
 			cond.Wait()
 		}
 		cond.L.Unlock()
-		if dlErr != nil {
-			return dlErr
-		}
 		if monitor.Enabled {
 			monitor.SegmentFullyTranscoded(nonce, seg.SeqNo, common.ProfilesNames(sess.Profiles), errCode)
 		}
+		if dlErr != nil {
+			return dlErr
```

Comment:
If download fails and retry succeeds, does it mean we get SegmentFullyTranscoded twice which means the success rate could be >100% ?

PM seems to have the same issue. Perhaps rather than moving this chunk of code, we should be moving the `SegmentFullyTranscoded` check to after PM and before we exit.

Reply:
Fixed